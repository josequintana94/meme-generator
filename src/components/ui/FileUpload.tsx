import React,{useRef} from "react";
import Button from "./Button";

interface IProps{
  btnText:string,
  handleChange:(event: any)=>void,
  className?:string,
}

export default function FileUpload({btnText,handleChange,className}:IProps) {

  const fileUploader = useRef<HTMLInputElement>(null);
  return (
    <>
      <input ref={fileUploader} style={{display: "none"}} type="file" onChange={handleChange}/>
      <Button
        id="upload"
        type="submit"
        className={className}
        onClick={() => fileUploader.current?.click()}
      >
        {btnText}
      </Button>
    </>
  );
  
}
