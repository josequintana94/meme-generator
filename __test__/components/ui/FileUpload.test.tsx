import renderer from "react-test-renderer";
import FileUpload from "../../../src/components/ui/FileUpload";

describe(">>>> Button", () => {
  test("++++ Snapshot Button", () => {
    const component = renderer.create(
      <FileUpload  
        btnText="File Upload"
        handleChange={jest.fn()}                     
      />             
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});

export {};