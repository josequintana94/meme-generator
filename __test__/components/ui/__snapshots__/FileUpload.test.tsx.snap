// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>>> Button ++++ Snapshot Button 1`] = `
[
  <input
    onChange={[MockFunction]}
    style={
      {
        "display": "none",
      }
    }
    type="file"
  />,
  <button
    id="upload"
    onClick={[Function]}
    type="submit"
  >
    File Upload
  </button>,
]
`;
